packages/runtime/src/module/index.ts

```diff
@@ -54,6 +54,10 @@ class Module {
       remoteEntryExports: this.remoteEntryExports,
       createScriptHook: (url: string) => {
         const res = this.loaderHook.lifecycle.createScript.emit({ url });
+        if (typeof document === 'undefined') {
+          //todo: needs real fix
+          return res as HTMLScriptElement;
```

Comment:
I don't quite understand why it's an h t m l script element without having to determine if res is empty

Reply:
I wasnt sure what type to set for the response that is a vm.runInthisContext() which returns the container

Reply:
https://github.com/module-federation/universe/pull/1741/files#diff-0d7aa9db8ded9ad201eeb48b22a2fd03110a2db035f29a4169696970a5a4f2db

See here how "createScript" works for node.js 
But the type is what, function? but if i do that, the types higher up break and im not sure theres alternative option to do this. Since we dont know what type remote is right now, check if in browser of not, then i use server version of script loading with VM